### PR TITLE
[BUG] Fix MaskedDataset y_masked initialized from x instead of y

### PR DESCRIPTION
**Pull request by kunal14901**
#### Reference Issues/PRs
Fixes a previously unreported bug in `MaskedDataset.__getitem__()`.

#### What does this implement/fix? Explain your changes.
`MaskedDataset.__getitem__()` in `pyaptamer/datasets/dataclasses/_masked.py` initializes `y_masked` by cloning `x` instead of `y`:

Before (line 156):
```python
x_masked = x.clone().detach()
y_masked = x.clone().detach()  # bug: should be y
```

After:
```python
x_masked = x.clone().detach()
y_masked = y.clone().detach()
```

When `x != y`, the masked target returned for loss computation contains values from the input sequence instead of the actual target sequence. The return docstring says it returns "target sequence with non-masked positions set to 0", but it was returning the input sequence with non-masked positions zeroed.

In typical MLM usage `x == y` so the bug is masked, but the class API explicitly accepts different `x` and `y` arrays via its constructor.

#### What should a reviewer concentrate their feedback on?
- Whether the fix correctly initializes `y_masked` from `y`
- Whether the regression test adequately covers the divergent x/y case

#### Did you add any tests for the change?
Yes. Added `pyaptamer/datasets/tests/test_masked_dataset.py` with 4 tests:
- `test_masked_dataset_length` — basic `__len__` check
- `test_masked_dataset_getitem_returns_four_tensors` — output shape
- `test_masked_dataset_mismatched_lengths` — `ValueError` on mismatched x/y
- `test_y_masked_initialized_from_y_not_x` — regression test for this specific bug

#### Any other comments?
Local validation:
- `pytest pyaptamer/datasets/tests/test_masked_dataset.py` — 4 passed
- `pre-commit run --files pyaptamer/datasets/dataclasses/_masked.py pyaptamer/datasets/tests/test_masked_dataset.py` — all passed

#### PR checklist
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG].
- [x] Added/modified tests
- [x] Used pre-commit hooks when committing to ensure that code is compliant with hooks.
```

### CODE REVIEW

**Comment by NennoMP** [comment]
The issue you are referring (#302) is not even open. It has been closed since cloning `y_masked` from `x` is expected behaviour as also observed in the authors' original code.

Closing this.

**Comment by kunal14901** [comment]
Thanks for the context, wasn't aware #302 was already closed as expected behaviour. Will check existing issues before opening PRs going forward.
